### PR TITLE
8343703: Symbol name cleanups after JEP 479

### PR DESCRIPTION
**Pull request by dholmes-ora**
After JEP 479 ([JDK-8339783](https://bugs.openjdk.org/browse/JDK-8339783) was integrated, the handling of certain symbol lookup code can be simplified. The old code needed to support 32-bit Windows, where names had a trailing `@<number>`. When this special case now is removed, some streamlining is possible. Specifically these arrays:

#define JNI_ONLOAD_SYMBOLS {"JNI_OnLoad"}
#define JNI_ONUNLOAD_SYMBOLS {"JNI_OnUnload"}
#define JVM_ONLOAD_SYMBOLS {"JVM_OnLoad"}
#define AGENT_ONLOAD_SYMBOLS {"Agent_OnLoad"}
#define AGENT_ONUNLOAD_SYMBOLS {"Agent_OnUnload"}
#define AGENT_ONATTACH_SYMBOLS {"Agent_OnAttach"}

are all singletons and so the actual strings can just be inlined directly into the code that uses them.

Testing:
- GHA
- Tiers 1-4 sanity
- 
Thanks

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8343703](https://bugs.openjdk.org/browse/JDK-8343703): Symbol name cleanups after JEP 479 (**Enhancement** - P4)

### Reviewers
 * [Alex Menkov](https://openjdk.org/census#amenkov) (@alexmenkov - **Reviewer**) Review applies to [c38bb48f](https://git.openjdk.org/jdk/pull/22380/files/c38bb48f865bbe600feb6b79eb0ded3e7934710d)
 * [Kim Barrett](https://openjdk.org/census#kbarrett) (@kimbarrett - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/22380/head:pull/22380` \
`$ git checkout pull/22380`

Update a local copy of the PR: \
`$ git checkout pull/22380` \
`$ git pull https://git.openjdk.org/jdk.git pull/22380/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 22380`

View PR using the GUI difftool: \
`$ git pr show -t 22380`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/22380.diff">https://git.openjdk.org/jdk/pull/22380.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/22380#issuecomment-2499787471)
</details>

### CODE REVIEW

**Comment by dholmes-ora** [comment]
Something is broken ...

**Comment by dholmes-ora** [comment]
Re-opened. There was a testing glitch caused by a change not part of this PR.

File: src/java.base/share/native/libjava/NativeLibraries.c [hunk 1]
```diff
@@ -64,45 +64,31 @@ static jboolean initIDs(JNIEnv *env)
...
-    if (isLoad) {
-        syms = onLoadSymbols;
-        symsLen = sizeof(onLoadSymbols) / sizeof(char *);
-    } else {
-        syms = onUnloadSymbols;
-        symsLen = sizeof(onUnloadSymbols) / sizeof(char *);
+    sym = isLoad ? "JNI_OnLoad" : "JNI_OnUnload";
+
+    // sym + '_' + cname + '\0'
+    if ((len = (cname != NULL ? (strlen(cname) + 1) : 0) + strlen(sym) + 1) >
```
Comment (alexmenkov) [inline src/java.base/share/native/libjava/NativeLibraries.c:76]: To match the updated comment:
```suggestion
    if ((len = strlen(sym) + (cname != NULL ? (strlen(cname) + 1) : 0) + 1) >
```

**Review by alexmenkov — APPROVED** [review]

**Comment by dholmes-ora** [comment]
Thanks for the review @alexmenkov !

**Review by alexmenkov — APPROVED** [review]

File: src/java.base/share/native/libjava/NativeLibraries.c [hunk 2]
```diff
@@ -64,45 +64,31 @@ static jboolean initIDs(JNIEnv *env)
...
-        }
+    jniFunctionName = malloc(len);
+    if (jniFunctionName == NULL) {
+      JNU_ThrowOutOfMemoryError(env, NULL);
+      goto done;
+    }
+    strcpy(jniFunctionName, sym);
+    if (cname != NULL) {
+      strcat(jniFunctionName, "_");
+      strcat(jniFunctionName, cname);
```
Comment (kimbarrett) [inline src/java.base/share/native/libjava/NativeLibraries.c:88]: Should be using 4-space indentation, rather than HotSpot's 2 space indentation.

**Review by kimbarrett — CHANGES_REQUESTED** [review]

File: src/java.base/share/native/libjava/NativeLibraries.c (hunk 2 above)
Comment (dholmes-ora) [reply src/java.base/share/native/libjava/NativeLibraries.c]: Good catch! Thanks for reviewing Kim.

**Review by kimbarrett — APPROVED** [review]
Looks good.

**Comment by dholmes-ora** [comment]
Thanks Kim!

/integrate

**Pull request closed without merging**